pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -81,11 +81,6 @@ public static ExpressionContext getExpression(Expression thriftExpression) {
    */
   public static FunctionContext getFunction(Function thriftFunction) {
     String functionName = thriftFunction.getOperator();
-    if (functionName.equalsIgnoreCase(AggregationFunctionType.COUNT.getName())) {
-      // NOTE: COUNT always take one single argument "*"
-      return new FunctionContext(FunctionContext.Type.AGGREGATION, AggregationFunctionType.COUNT.getName(),
-          new ArrayList<>(Collections.singletonList(ExpressionContext.forIdentifier("*"))));
-    }
```

Comment:
the reason why this starts to fail is b/c we removed these. can we keep this so the logic is correct?

Reply:
@walterddr In order to handle nulls in count function, we need to be able to accept two syntaxes: count(columnName) and count(*). We don't handle nulls when the input is count(*). 

I removed these lines and updated all calls of `CountAggregationFunction` constructor to send the count expression including the constructor calls in pinot-query-runtime. Please check `AggregateOperator.java` and `AggregationFunctionFactory.java` files. Only pinot-query-runtime still don't pass the argument passed in `CountAggregationFunction` constructor.

Reply:
yes. I definitely understand this. since `count(*)` by definition counts number of row since an entire row cannot be null; and `count(col)` only counts for non-null.

however since the multi-stage engine executes a logical planner after semantic parsing. `*` will always be resolved into a full list of table schema (and in the case of count(*) start will be dropped as the logical count doesn't take argument. see
https://github.com/apache/calcite/blob/b9c2099ea92a575084b55a206efc5dd341c0df62/core/src/main/java/org/apache/calcite/sql/SqlSyntax.java#L43-L47

so this is really the compatibility situation 
- since pinot current SQL parser only takes it to the semantic level it doesn't parse out the `*`, thus `*` remains a operand.
- but logically `*` means no argument in logical plan, thus in multi-stage engine it doesn't have argument.

Reply:
I would argument that since previously count with 0-operand and count with `*` identifier both work from RequestContextUtils. we should keep it this way (as jackie suggested in https://github.com/apache/pinot/pull/9086#discussion_r933876211). so no further action is needed here